pages/blog/publish-subscribe-semantics.md

```diff
@@ -0,0 +1,121 @@
+---
+> This post originally appeared on [IBM Integration Community](https://community.ibm.com/community/user/integration/blogs/nic-townsend1/2021/04/01/publish-vs-subscribe-in-asyncapi-and-why-it-may-no)
+
+As adoption of AsyncAPI increases, a frequent topic of conversation in the community is how to describe "publish" and "subscribe" semantics - both now and in future versions of the specification.
+
+In this blog post I aim to introduce the discussion and set you on the right path to document your event driven APIs.
+
+Or in other words, what code would you expect to be generated for the publish operation in the following AsyncAPI document?
+```
+    asyncapi: 2.0.0 
+      info:                   
+        title: My Application
+        version: 1.0 
+    servers:
+      bootstrap:
+        url: mybroker.com:3514
+        protocol: kafka
+    channels:
+      myChannel:
+        publish:
+          message:
+            payload:
+              type: string
+```  
+
+If you would be surprised to hear that this would result in a Kafka Consumer, then you should keep reading!
+
+![meme with phrase 'you keep using that word, I do not think you know what it means'](/img/posts/publish-subscribe-semantics/meme.webp)
+
+# In the beginning
+
+AsyncAPI started as an adaptation of OpenAPI - which describes synchronous "request/response" based APIs. In an OpenAPI world, you describe the application from the perspective of the client. Or in other words, the OpenAPI document describes how a client should interact with your application. The client and server communicate directly with each other.  
+  
+![server communicating with client](/img/posts/publish-subscribe-semantics/REST.webp)
+
+You document that a GET endpoint exists to access data, a POST endpoint exists to create data, etc. In all cases, a client speaks to the application (server) that is serving these endpoints.
+
+If you were implementing an application to honour the contract described in an OpenAPI document, you know to build route handlers that provide the documented endpoints for clients to access.
+
+_Side note – callbacks and webhooks are asynchronous operations, but in OpenAPI they are still described from the perspective of the client – the client has to initiate/register with the server before the server will push data to the client._
+
+# What about AsyncAPI?
+
+In an event driven architecture there is no client/server paradigm. Applications do not directly communicate with one another - instead, each application sends and receives events via communication channels provided by messaging infrastructure such as a broker. The broker ensures that events sent to a channel are delivered to interested applications. It can be considered "fire and forget" - an application sends an event, but does not have any interest in whether other applications receive or make use of the event.
```

Comment:
in general, I agree, but there is one exception that should be mentioned, it is websocket protocol. In websocket you still have client/server paradigm. maybe we can have just one sentence mentioning websocket as exception from this rule.

actually this is the origin of the confusion and change in the semantincs, in previous AsyncAPI version semantics were the other way around, but this was super confusing for websocket users because they follow client/server paradigm.